inference-engine/src/transformations/src/transformations/gru_cell_decomposition.cpp

```diff
@@ -0,0 +1,106 @@
+        auto Ht_R_zrh = std::make_shared<opset4::Split>(Ht_R, axis_1, 3);
+        auto biases_zrh = std::make_shared<opset4::Split>(B, axis_0, gru_cell->get_linear_before_reset() ? 4 : 3);
+
+        //  Xt*(Wz^T) + Ht-1*(Rz^T) + Wbz + Rbz
+        auto add_z_1 = std::make_shared<opset4::Add>(Ht_R_zrh->output(0), biases_zrh->output(0));
+        auto add_z_2 = std::make_shared<opset4::Add>(Xt_W_zrh->output(0), add_z_1);
+
+        // Xt*(Wr^T) + Ht-1*(Rr^T) + Wbr + Rbr
+        auto add_r_1 = std::make_shared<opset4::Add>(Ht_R_zrh->output(1), biases_zrh->output(1));
+        auto add_r_2 = std::make_shared<opset4::Add>(Xt_W_zrh->output(1), add_r_1);
+
+        auto clip = gru_cell->get_clip();
+        std::shared_ptr<Node> clamp_z = add_z_2;
+        std::shared_ptr<Node> clamp_r = add_r_2;
+        if (clip > 0.f) {
+            clamp_z = std::make_shared<opset4::Clamp>(add_z_2, -clip, clip);
+            clamp_r = std::make_shared<opset4::Clamp>(add_r_2, -clip, clip);
+            ngraph::copy_runtime_info(gru_cell, {clamp_z, clamp_r});
+        }
+
+        // zt = f(Xt*(Wz^T) + Ht-1*(Rz^T) + Wbz + Rbz)
+        auto z_t = ngraph::op::util::activation(gru_cell->get_activations()[0], clamp_z);
+        // rt = f(Xt*(Wr^T) + Ht-1*(Rr^T) + Wbr + Rbr)
+        auto r_t = ngraph::op::util::activation(gru_cell->get_activations()[0], clamp_r);
+
+        std::shared_ptr<Node>  _h;
+        if (gru_cell->get_linear_before_reset()) {
+            // _h = Xt*(Wh^T) + (rt (.) (Ht-1*(Rh^T) + Rbh)) + Wbh
+            auto Ht_Rh_Rbh = std::make_shared<opset4::Add>(Ht_R_zrh->output(2), biases_zrh->output(3));
+            auto mul_h_1 = std::make_shared<opset4::Multiply>(r_t, Ht_Rh_Rbh);
+            auto add_h_1 = std::make_shared<opset4::Add>(mul_h_1, biases_zrh->output(2));
+            _h = std::make_shared<opset4::Add>(Xt_W_zrh->output(2), add_h_1);
+            ngraph::copy_runtime_info(gru_cell, {Ht_Rh_Rbh, mul_h_1, add_h_1, _h});
+        } else {
+            // _h = Xt*(Wh^T) + (rt (.) Ht-1)*(Rh^T) + Rbh + Wbh
+            auto rt_Ht = std::make_shared<opset4::Multiply>(r_t, H_t);
+            auto mul_h_1 = std::make_shared<opset4::MatMul>(rt_Ht, R_zrh->output(2), false, true);
+            auto add_h_1 = std::make_shared<opset4::Add>(mul_h_1, biases_zrh->output(2));
+            _h = std::make_shared<opset4::Add>(Xt_W_zrh->output(2), add_h_1);
+            ngraph::copy_runtime_info(gru_cell, {rt_Ht, mul_h_1, add_h_1, _h});
+        }
+        // ht = g(_h)
+        std::shared_ptr<Node> clamp_h = _h;
+        if (clip > 0.f) {
+            clamp_h = std::make_shared<opset4::Clamp>(_h, -clip, clip);
+            ngraph::copy_runtime_info(gru_cell, clamp_h);
+        }
+        auto h_t = ngraph::op::util::activation(gru_cell->get_activations()[1], clamp_h);
+
+        // Ht = (1 - zt) (.) ht + zt (.) Ht-1
+        auto one = opset4::Constant::create(z_t->get_element_type(),
+                                            z_t->get_shape(),
```

Comment:
You need to check that shape is static. You can filter out dynamic cases by using pattern predicate that checks output shape. `pattern::has_static_shape`